src/openMVG/sfm/pipelines/sequential/sequential_SfM.cpp

```diff
@@ -418,120 +537,145 @@ bool SequentialSfMReconstructionEngine::AutomaticInitialPairChoice(Pair & initia
-            std::make_pair(cam_I->w(), cam_I->h()), std::make_pair(cam_J->w(), cam_J->h()),
-            256) && relativePose_info.vec_inliers.size() > iMin_inliers_count)
-          {
-            // Triangulate inliers & compute angle between bearing vectors
-            std::vector<float> vec_angles;
-            vec_angles.reserve(relativePose_info.vec_inliers.size());
-            const Pose3 pose_I = Pose3(Mat3::Identity(), Vec3::Zero());
-            const Pose3 pose_J = relativePose_info.relativePose;
-            const Mat34 PI = cam_I->get_projective_equivalent(pose_I);
-            const Mat34 PJ = cam_J->get_projective_equivalent(pose_J);
-            for (const size_t inlier_idx : relativePose_info.vec_inliers)
-            {
-              Vec3 X;
-              TriangulateDLT(PI, xI.col(inlier_idx), PJ, xJ.col(inlier_idx), &X);
-
-              openMVG::tracks::STLMAPTracks::const_iterator iterT = map_tracksCommon.begin();
-              std::advance(iterT, inlier_idx);
-              tracks::submapTrack::const_iterator iter = iterT->second.begin();
-              const Vec2 featI = _features_provider->feats_per_view[I][iter->second].coords().cast<double>();
-              const Vec2 featJ = _features_provider->feats_per_view[J][(++iter)->second].coords().cast<double>();
-              vec_angles.push_back(AngleBetweenRay(pose_I, cam_I, pose_J, cam_J, featI, featJ));
-            }
-            // Compute the median triangulation angle
-            const unsigned median_index = vec_angles.size() / 2;
-            std::nth_element(
-              vec_angles.begin(),
-              vec_angles.begin() + median_index,
-              vec_angles.end());
-            const float scoring_angle = vec_angles[median_index];
-            // Store the pair iff the pair is in the asked angle range [fRequired_min_angle;fLimit_max_angle]
-            if (scoring_angle > fRequired_min_angle &&
-                scoring_angle < fLimit_max_angle)
-            {
-  #ifdef OPENMVG_USE_OPENMP
-              #pragma omp critical
-  #endif
-              scoring_per_pair.emplace_back(scoring_angle, current_pair);
-            }
-          }
-        }
+    if (relativePoseSuccess && relativePose_info.vec_inliers.size() > iMin_inliers_count)
+    {
+      // Triangulate inliers & compute angle between bearing vectors
+      std::vector<float> vec_angles(relativePose_info.vec_inliers.size());
+      std::vector<std::size_t> validCommonTracksIds(relativePose_info.vec_inliers.size());
+      const Pose3 pose_I = Pose3(Mat3::Identity(), Vec3::Zero());
+      const Pose3 pose_J = relativePose_info.relativePose;
+      const Mat34 PI = cam_I->get_projective_equivalent(pose_I);
+      const Mat34 PJ = cam_J->get_projective_equivalent(pose_J);
+      std::size_t i = 0;
+      for (const size_t inlier_idx: relativePose_info.vec_inliers)
+      {
+        Vec3 X;
+        TriangulateDLT(PI, xI.col(inlier_idx), PJ, xJ.col(inlier_idx), &X);
```

Comment:
I think we should check whether X is in front of the cameras, and if not reject it (not putting it into `vec_angles `)